src/StructureFactors/SampleGeneration.jl

```diff
@@ -113,12 +113,16 @@ function accum_sample!(sf::StructureFactor)
     nsamples[1] += 1
 
     # Transfer to final memory layout while accumulating new samplebuf
-    for ω in 1:nω, cell in CartesianIndices(latsize), j in 1:natoms, i in 1:natoms
+    cache = zeros(ComplexF64, size(data)[4:7])
+    for j in 1:natoms
+      for i in 1:natoms
         for (ci, c) in idxinfo 
-            α, β = ci.I
-            old = data[c,i,j,cell,ω]
-            data[c,i,j,cell,ω] = old + (samplebuf[α,cell,i,ω] * conj(samplebuf[β,cell,j,ω]) - old) / nsamples[1]
```

Comment:
There might be a semantic difference in the way that the running average had been calculated. By subtracting the mean (`old`) and then adding it back, it the previous version could possibly be reducing floating point round-off error. See here for an example: https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Online . It's not clear if this difference in floating point roundoff is important for this context.

Reply:
The semantics and algorithm are the same, it's just in two steps with explicit copy buffer instead of one step with implicit (allocated on demand) copy buffer.

To see that the copy buffer is required, note that the result of the first step (computing `x_n - \bar x_{n-1}`) must be stored somewhere and, at the same time, the original `\bar x_{n-1}` must be stored somewhere else.

The alternative to the copy buffer is to to use `samplebuf[α,:,:,:,i,:]` as the copy buffer, but this clobbers the `samplebuf`. But that might be OK, I'm not sure what Sunny's semantics/assumptions about that are!